### PR TITLE
Recaptcha Admin SDK related changes.

### PR DESCRIPTION
**Pull request by Xiaoshouzi-gh**
1. Update/Get recaptcha config of project.
2. Update/Get recaptcha config of a tenant.

This branch should NOT be merged to master after the review.

### Discussion

  * Read the contribution guidelines (CONTRIBUTING.md).
  * If this has been discussed in an issue, make sure to link to the issue here. 
    If not, go file an issue about this **before creating a pull request** to discuss.

### Testing

  * Make sure all existing tests in the repository pass after your change.
  * If you fixed a bug or added a feature, add a new test to cover your code.

### API Changes

  * At this time we cannot accept changes that affect the public API.  If you'd like to help 
    us make Firebase APIs better, please propose your change in an issue so that we 
    can discuss it together.

### CODE REVIEW

File: src/auth/project-config-manager.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import {
+  AuthRequestHandler,
+} from './auth-api-request';
+
+/**
+ * Defines the project config manager used to help manage project config related operations.
+ * This includes:
+ * <ul>
+ * <li>The ability to update and get project config.</li>
+ */
```
Comment (kevinthecheung) [inline src/auth/project-config-manager.ts:27]:
```suggestion
 * Manages (gets and updates) the current project config.
 */
```

File: src/auth/project-config.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+   * agreeing to reCAPTCHA Enterprise
+   * {@link https://cloud.google.com/terms/service-terms | Term of Service}.
+   */
+  recaptchaConfig?: RecaptchaConfig;
+}
+
+/**
+ * Response received from get/update project config.
+ * We are only exposing the recaptcha config for now.
+ */
```
Comment (kevinthecheung) [inline src/auth/project-config.ts:39]:
```suggestion
/**
 * Response received when getting or updating the project config.
 * Currently only includes the reCAPTCHA config.
 */
```

File: src/auth/project-config.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+export interface ProjectConfigServerResponse {
+  recaptchaConfig?: RecaptchaConfig;
+}
+
+/**
+ * Request sent to update project config.
+ * We are only updating the recaptcha config for now.
+ */
```
Comment (kevinthecheung) [inline src/auth/project-config.ts:47]:
```suggestion
/**
 * Request to update the project config.
 * Currently only includes the reCAPTCHA config.
 */
```

File: src/auth/project-config.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+   * 
+   * @internal
+   */
+  public static buildServerRequest(configOptions: UpdateProjectConfigRequest): ProjectConfigClientRequest {
+    ProjectConfig.validate(configOptions);    
+    return configOptions as ProjectConfigClientRequest;
+  }
+ 
+  /**
+   * The recaptcha configuration.
```
Comment (kevinthecheung) [inline src/auth/project-config.ts:106]:
```suggestion
   * The reCAPTCHA configuration.
```

File: src/auth/project-config.ts [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+/**
+* Represents a project configuration.
+*/
+export class ProjectConfig {
+  /**
+   * The recaptcha configuration to update on the project config.
+   * By enabling reCAPTCHA Enterprise Integration you are
+   * agreeing to reCAPTCHA Enterprise
+   * {@link https://cloud.google.com/terms/service-terms | Term of Service}.
+   */
```
Comment (kevinthecheung) [inline src/auth/project-config.ts:61]:
```suggestion
  /**
   * The reCAPTCHA configuration to update on the project.
   * By enabling reCAPTCHA Enterprise integration, you are
   * agreeing to the reCAPTCHA Enterprise
   * {@link https://cloud.google.com/terms/service-terms | Term of Service}.
   */
```

**Review by kevinthecheung — APPROVED** [review]

**Comment by Xiaoshouzi-gh** [comment]
Thank you Kevin!  Feedback are address in this PR https://github.com/firebase/firebase-admin-node/pull/1685 and will be merge into the `recaptcha` branch.  Closing this PR.

**Pull request closed without merging**